### DISCUSSION TITLE
Accessibility: How to support or detect screen readers

### DISCUSSION THREAD

**Discussion by Josverl** [Textualize/rich#3353] [General] [answered]
Hi there,

Ive been using rich and rich-click in several of the CLI tools that I create , and based on feedback from users I started check how screen readers such as Windows narrator can read the output and updates from the CLI tools.

One of the things that i found wat that the progress tracker is '**_rather_**' annoying as it tries to keep up with the updates.
The easy fix I found was to lower the update frequency on all progress trackers to have a balance between visual pleasantness and audible accessibility

But that has made me wonder if there are other options, where rich can help by perhaps autodetecting screen readers or similar ?
Is there such a thing as an audio accessible theme perhaps ?

**Comment by Josverl** [comment, marked as answer]
Closing for lack of response or interest.

**Reply by willmcgugan** [reply]
I know of no way to detect screen readers or any protocol to work with them in the terminal. It is a sad state of affairs frankly.
